terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/runners.ts

```diff
@@ -963,3 +881,110 @@ export async function createRunner(runnerParameters: RunnerInputParameters, metr
     throw e;
   }
 }
+
+function isRunnerReusable(runner: RunnerInfo, useCase: string): boolean {
```

Comment:
return a reasoning enum, and maybe a text for logging, and do the logging in the function invocating it.

example of the idea:
```
return {
  decision: ReuseStatusDecision.NoReuse_NoGithubRunnerIDTag,
  message: `Runner ${runner.instanceId} does not have a GithubRunnerID tag`,
};
```